addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/internal/handler/MySensorsBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+import org.openhab.binding.mysensors.internal.factory.MySensorsCacheFactory;
+import org.openhab.binding.mysensors.internal.protocol.MySensorsBridgeConnection;
+import org.openhab.binding.mysensors.internal.protocol.ip.MySensorsIpConnection;
+import org.openhab.binding.mysensors.internal.protocol.serial.MySensorsSerialConnection;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsDeviceManager;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * MySensorsBridgeHandler is used to initialize a new bridge (in MySensors: Gateway)
+ * The sensors are connected via the gateway/bridge to the controller
+ *
+ * @author Tim Oberföll
+ *
+ */
+public class MySensorsBridgeHandler extends BaseBridgeHandler implements MySensorsUpdateListener {
+
+    private Logger logger = LoggerFactory.getLogger(MySensorsBridgeHandler.class);
+
+    // Bridge connection
+    private MySensorsBridgeConnection myCon = null;
+
+    // Device manager
+    private MySensorsDeviceManager myDevManager = null;
+
+    // Configuration from thing file
+    private MySensorsBridgeConfiguration myConfiguration = null;
+
+    // Cache file
+    private MySensorsCacheFactory bindingCacheFile = null;
+
+    public MySensorsBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initialization of the MySensors bridge");
+
+        myConfiguration = getConfigAs(MySensorsBridgeConfiguration.class);
+
+        if (getThing().getThingTypeUID().equals(THING_TYPE_BRIDGE_SER)) {
+            myCon = new MySensorsSerialConnection(this, myConfiguration.serialPort, myConfiguration.baudRate,
+                    myConfiguration.sendDelay);
+        } else if (getThing().getThingTypeUID().equals(THING_TYPE_BRIDGE_ETH)) {
+            myCon = new MySensorsIpConnection(this, myConfiguration.ipAddress, myConfiguration.tcpPort,
+                    myConfiguration.sendDelay);
+        } else {
+            logger.error("Not recognized bridge: {}", getThing().getThingTypeUID());
+        }
+
+        if (myCon != null) {
+            myCon.initialize();
+            myCon.addEventListener(this);
+        }
+
+        myDevManager = new MySensorsDeviceManager(myCon, loadCacheFile());
+        myCon.addEventListener(myDevManager);
+
+        logger.debug("Initialization of the MySensors bridge DONE!");
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing of the MySensors bridge");
+        if (myCon != null) {
+            myCon.removeEventListener(myDevManager);
+            myCon.removeEventListener(this);
+            myCon.destroy();
+        }
+
+        saveCacheFile();
+
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
```

Comment:
The abstract method handleCommand() needs to be implemented. That's what eclipse tells me. :)